packages/ui/src/components/va-counter/VaCounter.vue

```diff
@@ -0,0 +1,391 @@
+
+    const {
+      isFocused,
+      // will be useful when we resolve problem with 'withConfigTransport'
+      focus,
+      blur,
+    } = useFocus(input, emit)
+
+    const { valueComputed } = useStateful(props, emit)
+
+    const setCountInput = ({ target } : { target: HTMLInputElement }) => {
+      valueComputed.value = Number(target?.value)
+    }
+
+    const setCountChange = ({ target } : { target: HTMLInputElement }) => {
+      calculateCounterValue(Number(target?.value))
+    }
+
+    const getRoundDownWithStep = (value: number) => {
+      if (!props.min || !props.step) { return value }
+      return props.min + props.step * Math.floor((value - props.min) / props.step)
+    }
+
+    const calculateCounterValue = (counterValue: number) => {
+      if (props.min && counterValue < props.min) {
+        valueComputed.value = props.min
+        return
+      }
+
+      if (props.max && (counterValue > props.max)) {
+        valueComputed.value = getRoundDownWithStep(props.max)
+        return
+      }
+
+      valueComputed.value = getRoundDownWithStep(counterValue)
+    }
+
+    const isMinReached = computed(() => {
+      if (!props.min) { return false }
+      return Number(valueComputed.value) <= props.min
+    })
+
+    const isMaxReached = computed(() => {
+      if (!props.max) { return false }
+
+      return props.step
+        ? Number(valueComputed.value) > (props.max - props.step)
+        : Number(valueComputed.value) >= props.max
+    })
+
+    const disabledDecreaseAction = computed(() => (
+      isMinReached.value || props.readonly || props.disabled
+    ))
+
+    const disabledIncreaseAction = computed(() => (
+      isMaxReached.value || props.readonly || props.disabled
+    ))
+
+    const decreaseCount = () => {
+      if (disabledDecreaseAction.value) { return }
+      calculateCounterValue(Number(valueComputed.value) - props.step)
+    }
+
+    const increaseCount = () => {
+      if (disabledIncreaseAction.value) { return }
+      calculateCounterValue(Number(valueComputed.value) + props.step)
+    }
+
+    const { colorComputed } = useColor(props)
+
+    const decreaseIconProps = computed(() => ({
+      class: { 'va-counter__icon--inactive': disabledDecreaseAction.value },
+      color: colorComputed.value,
+      name: props.decreaseIcon,
+      ...(!disabledDecreaseAction.value && { onClick: decreaseCount }),
+    }))
+
+    const increaseIconProps = computed(() => ({
+      class: { 'va-counter__icon--inactive': disabledIncreaseAction.value },
+      color: colorComputed.value,
+      name: props.increaseIcon,
+      ...(!disabledIncreaseAction.value && { onClick: increaseCount }),
+    }))
+
+    const squareCorners = computed(() => (
+      (typeof props.margins === 'string' ? parseFloat(props.margins) : props.margins) === 0
+    ))
+
+    const decreaseButtonProps = computed(() => ({
+      ...pick(props, ['outline', 'rounded', 'color', 'textColor']),
```

Comment:
Why decreaseButtonProps and increaseButtonProps are different?

Reply:
They have different computed `disabled` attr.
Refacrored a bit

Reply:
> They have different computed `disabled` attr. Refacrored a bit

Oh. I see you forget to remove 'outline' from pick, because it used a bit down

```ts
      outline: props.flat && props.outline,
```